src/CodingContractGenerator.ts

```diff
@@ -14,6 +14,27 @@ import { BaseServer } from "./Server/BaseServer";
 
 import { getRandomIntInclusive } from "./utils/helpers/getRandomIntInclusive";
 import { ContractFilePath, resolveContractFilePath } from "./Paths/ContractFilePath";
+import { clampNumber } from "./utils/helpers/clampNumber";
+
+export function tryGeneratingRandomContract(numberOfTries: number): void {
+  // We try to generate a contract every 10 minutes. 525600 is the number of tries in 10 years. There is no reason to
+  // support anything above that.
+  numberOfTries = clampNumber(Math.floor(numberOfTries), 0, 525600);
+  if (numberOfTries < 1) {
+    return;
+  }
+  let currentNumberOfContracts = GetAllServers().reduce((sum, server) => {
+    return sum + server.contracts.length;
+  }, 0);
+  for (let i = 0; i < numberOfTries; ++i) {
+    const random = Math.random();
+    if (random > 0.25 / (1 + Math.exp(0.0012 * currentNumberOfContracts - 6))) {
```

Comment:
After analysis, I have a slightly different set of constants to suggest: `100 / (399 + Math.exp(0.0012 * currentNumberOfContracts))`. Primarily to exactly match the probability/slope of 0.25 at 0 contracts.

Analysis also reveals that there *is* no asymptote, the behavior is logarithmic. IMO this is a good thing, it's what I was asking for.

https://www.desmos.com/calculator/zmpohx9eps